dynamic_programming/minimum_palindrome_partitioning.cpp

```diff
@@ -0,0 +1,91 @@
+/**
+ * @file
+ * @brief Program to find the Minimum Palindrome
+ * Partitioning of a string
+ *
+ * @details
+ * [Palindrome](https://en.wikipedia.org/wiki/Palindrome) string is sequence
+ * of characters which reads the same backward as forward
+ * [Substring](https://en.wikipedia.org/wiki/Substring) is a contiguous sequence
+ * of characters within a string
+ * Partitioning of a string is palindrome partitioning if every substring
+ * of the partition is a palindrome.
+ * This code will calculate the minimum number of partitions in a string
+ * such that all substrings are palindrome.
+
+ * @author [Anjali Jha](https://github.com/anjali1903)
+ */
+
+#include <algorithm>
+#include <cassert>
+#include <iostream>
+#include <vector>
+
+/**
+ * Function that returns the minimum palindrome
+ * partitioning of a string
+ */
+int mpp(std::string s) {  // mpp: minimum palindrome partitioning
```

Comment:
pass by reference instead of creating a copy value. 
```suggestion
int mpp(const std::string &s) {  // mpp: minimum palindrome partitioning
```